src/analysis/processing/qgsalgorithmconcavehullbyfeature.cpp

```diff
@@ -0,0 +1,146 @@
+    email                : nyall dot dawson at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmconcavehullbyfeature.h"
+#include "qgsgeometrycollection.h"
+
+///@cond PRIVATE
+
+QString QgsConcaveHullByFeatureAlgorithm::name() const
+{
+  return QStringLiteral( "concavehullbyfeature" );
+}
+
+QString QgsConcaveHullByFeatureAlgorithm::displayName() const
+{
+  return QObject::tr( "Concave hull (by feature)" );
+}
+
+QStringList QgsConcaveHullByFeatureAlgorithm::tags() const
+{
+  return QObject::tr( "concave,hull,bounds,bounding" ).split( ',' );
+}
+
+QString QgsConcaveHullByFeatureAlgorithm::group() const
+{
+  return QObject::tr( "Vector geometry" );
+}
+
+QString QgsConcaveHullByFeatureAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeometry" );
+}
+
+QString QgsConcaveHullByFeatureAlgorithm::outputName() const
+{
+  return QObject::tr( "Concave hulls" );
+}
+
+QString QgsConcaveHullByFeatureAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "This algorithm calculates the concave hull for each feature in an input layer." ) + QStringLiteral( "\n\n" )
+         + QObject::tr( "A concave hull is a polygon which contains all the points of the input geometries, but is a better approximation than the convex hull to the area occupied by the input." ) + QStringLiteral( "\n\n" )
+         + QObject::tr( "It is frequently used to convert a multi-point into a polygonal area which contains all the points from the input geometry." ) + QStringLiteral( "\n\n" )
+         + QObject::tr( "See the 'Concave hull (by layer)' algorithm for a concave hull calculation which covers the whole layer or grouped subsets of features." );
+}
+
+QString QgsConcaveHullByFeatureAlgorithm::shortDescription() const
+{
+  return QObject::tr( "Calculates the concave hull for each feature in an input layer." );
+}
+
+QgsConcaveHullByFeatureAlgorithm *QgsConcaveHullByFeatureAlgorithm::createInstance() const
+{
+  return new QgsConcaveHullByFeatureAlgorithm();
+}
+
+void QgsConcaveHullByFeatureAlgorithm::initParameters( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterNumber( QStringLiteral( "ALPHA" ), QObject::tr( "Threshold (0-1, where 1 is equivalent with Convex Hull)" ), Qgis::ProcessingNumberParameterType::Double, 0.3, false, 0, 1 ) );
```

Comment:
Sorry for the noob question: there is no "input layer" param in this alg? Or (for my own understanding), is it subtly set in the next method? Thanks

Reply:
It is a feature-based alg, input is set automatically.